graphql-dgs-spring-boot-micrometer/src/main/kotlin/com/netflix/graphql/dgs/metrics/micrometer/GraphQLMetricsInstrumentation.kt

```diff
@@ -0,0 +1,108 @@
+import io.micrometer.core.instrument.Timer
+
+import org.springframework.boot.actuate.metrics.AutoTimer
+import java.util.*
+import java.util.concurrent.CompletableFuture
+
+class GraphQLMetricsInstrumentation(
+        private val registry: MeterRegistry,
+        private val tagsProvider: GraphQLTagsProvider,
+        private val autoTimer: AutoTimer
+) : SimpleInstrumentation() {
+
+    override fun createState(): InstrumentationState {
+        return MetricsInstrumentationState(this.registry)
+    }
+
+    override fun beginExecution(parameters: InstrumentationExecutionParameters): InstrumentationContext<ExecutionResult> {
+        val state: MetricsInstrumentationState = parameters.getInstrumentationState()
+        state.startTimer()
+
+        return object : SimpleInstrumentationContext<ExecutionResult>() {
+            override fun onCompleted(result: ExecutionResult, exc: Throwable?) {
+                state.stopTimer(autoTimer.builder("gql.query")
+                        .tags(tagsProvider.getExecutionTags(parameters, result, exc))
+                        .tags(tagsProvider.getEnvironmentTags()))
+            }
+        }
+    }
+
+    override fun instrumentExecutionResult(executionResult: ExecutionResult, parameters: InstrumentationExecutionParameters): CompletableFuture<ExecutionResult> {
+        sanitizeErrorPathsForTags(executionResult).forEach { _ ->
+            registry.counter("gql.error",
+                    Tags.of(tagsProvider.getEnvironmentTags()).and(tagsProvider.getExecutionTags(parameters, executionResult, null)))
+                    .increment()
+        }
+        return CompletableFuture.completedFuture(executionResult)
+    }
+
+    override fun instrumentDataFetcher(dataFetcher: DataFetcher<*>, parameters: InstrumentationFieldFetchParameters): DataFetcher<*> {
+        val gqlField = InstrumentationTags.findDataFetcherTag(parameters)
+        if (parameters.isTrivialDataFetcher || InstrumentationTags.shouldIgnored(gqlField)) {
+            return dataFetcher
+        }
+
+        return DataFetcher { environment ->
+            val tags = Tags.of("gql.field", gqlField).and(tagsProvider.getFieldFetchTags(parameters)).and(tagsProvider.getEnvironmentTags())
+            registry.counter("gql.resolver.count", tags).increment()
+
+            val sampler = Timer.start(registry)
+            val result = dataFetcher.get(environment)
+            if (result is CompletableFuture<*>) {
+                result.thenAccept {
+                    recordDataFetcherTime(sampler, tags, registry)
+                }
+            } else {
+                recordDataFetcherTime(sampler, tags, registry)
+            }
+            result
+        }
+    }
+
+    private fun recordDataFetcherTime(timerSampler: Timer.Sample, tags: Tags, registry: MeterRegistry) {
+        timerSampler.stop(registry,
+                Timer.builder("gql.resolver.time").tags(tags).publishPercentileHistogram())
```

Comment:
Suggest striking `publishPercentileHistogram` as it it can be turned on later by an application developer if they want histograms for this timer in application.properties with `management.metrics.distribution.percentiles-histogram.gql=true`

Reply:
had no idea, thanks for the pointer.